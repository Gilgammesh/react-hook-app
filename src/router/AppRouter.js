import React from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import AboutScreen from "../components/09-useContext/AboutScreen";
import HomeScreen from "../components/09-useContext/HomeScreen";
import LoginScreen from "../components/09-useContext/LoginScreen";
import Screen404 from "../components/09-useContext/Screen404";

const AppRouter = () => {
  return (
    <BrowserRouter>
      <div>
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route exact path="/home" component={HomeScreen} />
          <Route exact path="/about" component={AboutScreen} />
          <Route exact path="/login" component={LoginScreen} />
          <Route component={Screen404} />
        </Switch>
      </div>
    </BrowserRouter>
  );
};

export default AppRouter;
