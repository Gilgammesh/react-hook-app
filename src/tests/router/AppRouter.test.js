import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { mount } from "enzyme";
import "@testing-library/jest-dom";
import AppRouter from "../../router/AppRouter";
import UserContext from "../../components/09-useContext/UserContext";

describe("Pruebas unitarias en el componente AppRouter.js", () => {
  const user = {
    id: 1,
    name: "Carlos",
    email: "carlos.santander@gmail.com",
    password: "123456",
  };
  const setUser = jest.fn();
  /************************************************************************************/
  test("se debe mostrar el componente <AppRouter /> correctamente", () => {
    const wrapper = mount(
      <UserContext.Provider
        value={{
          user,
          setUser,
        }}
      >
        <AppRouter />
      </UserContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
