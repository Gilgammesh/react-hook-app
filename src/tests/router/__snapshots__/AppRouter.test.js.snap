// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente AppRouter.js se debe mostrar el componente <AppRouter /> correctamente 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-primary"
    >
      <div
        className="navbar-collapse"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/home"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </ul>
      </div>
    </nav>,
    <div
      className="m-4"
    >
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Carlos",
   "email": "carlos.santander@gmail.com",
   "password": "123456"
}
      </pre>
    </div>,
  ]
</div>
`;
